examples/text-replacer/popup.js

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+const form = document.querySelector('form');
+
+load(['patterns'])
+  .then(data => data.patterns)
+  .then(loadFormData);
+
+form.addEventListener('submit', async (event) => {
+  event.preventDefault();
+  await saveFormData();
+
+  let currentTab = await getCurrentTab();
+  chrome.scripting.executeScript({
+    target: {tabId: currentTab.id},
+    files: ['content.js']
+  });
+});
+
+document.getElementById('clear').addEventListener('click', (event) => {
```

Comment:
You can get rid of this and change the `<button>` to be an `<input type="reset" />`.